xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -248,6 +194,209 @@ private static ManagedChannel initLbChannel(
     return channel;
   }
 
+  /**
+   * A LoadBalancerProvider that provides a load balancer with a fixed XdsConfig.
+   */
+  private final class FixedXdsConfigBalancerProvider extends LoadBalancerProvider {
+    final XdsConfig xdsConfig;
+    final XdsConfig oldXdsConfig;
+    final EndpointWatcher oldEndpointWatcher;
+
+    FixedXdsConfigBalancerProvider(XdsConfig xdsConfig) {
+      this.xdsConfig = xdsConfig;
+      oldXdsConfig = LookasideLb.this.xdsConfig;
+      oldEndpointWatcher = endpointWatcher;
+    }
+
+    @Override
+    public boolean isAvailable() {
+      return true;
+    }
+
+    @Override
+    public int getPriority() {
+      return 5;
+    }
+
+    // A synthetic policy name identified by xds config.
+    @Override
+    public String getPolicyName() {
+      return "xds_policy__balancer_name_" + xdsConfig.balancerName
+          + "__childPolicy_" + xdsConfig.childPolicy
+          + "__fallbackPolicy_" + xdsConfig.fallbackPolicy
+          + "__edsServiceName_" + xdsConfig.edsServiceName
+          + "__lrsServerName_" + xdsConfig.lrsServerName;
+    }
+
+    @Override
+    public LoadBalancer newLoadBalancer(Helper helper) {
+      return new FixedXdsConfigBalancer(helper);
+    }
+
+    final class FixedXdsConfigBalancer extends LoadBalancer {
+      final Helper helper;
+
+      // All fields become non-null once handleResolvedAddresses() successfully.
+      // All fields are assigned at most once.
+      @Nullable
+      ObjectPool<XdsClient> xdsClientRef;
+      @Nullable
+      XdsClient xdsClient;
+      @Nullable
+      LocalityStore localityStore;
+      @Nullable
+      LoadReportClient lrsClient;
+
+      FixedXdsConfigBalancer(Helper helper) {
+        this.helper = helper;
+      }
+
+      @Override
+      public void handleNameResolutionError(Status error) {}
+
+      @Override
+      public boolean canHandleEmptyAddressListFromNameResolution() {
+        return true;
+      }
+
+      @Override
+      public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+        if (xdsClientRef != null) {
+          return;
+        }
```

Comment:
Not clear about why you are returning here: is this to avoid the recursion case? A comment would be helpful then.

Reply:
Yes, I also feel this logic is confusing. But what he wants to convey is the idea that each different `XdsConfig` (by different, I mean at least `childPolicy` or `edsServiceName` is different) should have its own balancer implementation. So its `handleResolvedAddresses()` is expected to only be invoked at most once per XdsConfig-fixed balancer. It is indeed very awkward. But unless the implementor can come up with a new idea/design to handle switch internally, seems not much we can do...

Reply:
`xdsClientRef` is a member variable that is set later in this method for the 3 different cases described in the comment on line 280. The only way this code kicks in is with recursion happening. For the different XdsConfig issue @voidzcy mentioned the code on lines 131-135 takes care of it. So it will be good to have a clarifying comment here,.

Reply:
As mentioned in the comments
```
      // All fields become non-null once handleResolvedAddresses() successfully.
      // All fields are assigned at most once.
      @Nullable
      ObjectPool<XdsClient> xdsClientRef;
```

These fields are only assigned one, and once assigned, `handleResolvedAddresses()` should be no-op because the config is fixed in this balancer.

Reply:
> But unless the implementor can come up with a new idea/design to handle switch internally, seems not much we can do...

@ejona86 is strongly against handling switch internally because that is hard to test, and so easy to introduce bugs, and because switching is rarely executed in production, a bug could be not noticed for quite a while.

Reply:
>So it will be good to have a clarifying comment here,.

Added clarifying comment.